Rework of TS07

fabianoboril opened the pull request:
Completed rework of scenario 07 according to feedback. We did not use a stream of adversaries for now, to increase the challenge of this scenario, but we modified the behavior to ensure a collision for various driving behaviors.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/carla-simulator/scenario_runner/107)
<!-- Reviewable:end -->

mohit-awana commented on srunner/scenarios/opposite_vehicle_taking_priority.py:
```diff
@@ -133,24 +133,44 @@ def _create_behavior(self):
-        keep_velocity_for_distance = py_trees.composites.Parallel(
-            "Keep velocity for distance",
+        # Generate plan for WaypointFollower
+        waypoint = self.other_actors[0].get_world().get_map().get_waypoint(self.other_actors[0].get_location())
```
get_map() is now available in CarlaDataProvider.

fabianoboril replied:
Fixed.

mohit-awana commented on srunner/scenarios/opposite_vehicle_taking_priority.py:
```diff
@@ -133,24 +133,44 @@ def _create_behavior(self):
+        continue_driving_timeout = TimeOut(10)
+
+        continue_driving.add_child(continue_driving_waypoints)
+        continue_driving.add_child(continue_driving_distance)
```
Currently, if the ego vehicle decides to drive in the same direction as adversary, the ego vehicle is stuck behind the adversary vehicle, it has to change lane to complete the scenario. Is it intended? Otherwise scenario ends with a timeout as the end condition of the scenario doesn't meet.

fabianoboril replied:
Yes correct. The only chance I see to avoid this, would be to feed the ego route into the scenario. This however, would require some re-design.

For now, I think we can still use this scenario. Even if the adversary takes the same route, it will be in front of the ego vehicle (so a follow maneuver), and disappear as soon as this scenario is over. The disappearance could be made even a bit more aggressively.

So, all in all, my suggestion would be to keep it as it is for now. If our testing reveals that it is a real problem, we can come back and work on it.

fabianoboril commented:
@mohit-awana: Can we merge it?

mohit-awana commented:
> @mohit-awana: Can we merge it?

Okay, yes please :D

mohit-awana commented:
Sorry, my bad :D Pressed the wrong button :P

fabianoboril commented:
> > ```
> > OppositeVehicleRunningRedLight011
> > ```
> 
> I am not getting the expected behavior. Have a look:
> 
> https://youtu.be/vCH2CsEMj-Y
> 
> > _[Reviewable](https://reviewable.io/reviews/carla-simulator/scenario_runner/107#-:-Lb9g3RH1Ib0VhmywJiW:b-mcxzk1)_ status: 0 of 10 files reviewed, 2 unresolved discussions (waiting on @germanros1987 and @mohit-awana)

Thanks for the video. The new default-behavior of avoiding collisions of the WaypointFollower caused the other actor to not move. I fixed this and also updated the behavior slightly to improve the scenario for low ego velocities.

fabianoboril commented:
Rebase done :)
